### ISSUE TITLE
Feature Request / Issue: Doesn't use $(Build.SourceVersion) on sourceBranch when creating PR

### ISSUE THREAD

**Issue by ColinKrist** [shayki5/azure-devops-create-pr-task#87]
Our team uses a trunk based deployment flow and we don't cut release/* branches for release -> master merges which I expect most users of this task do. We expected the commit tied to the Build on the YAML Build Pipeline to be used as the last commit to the `sourceBranch`; however, it looks like in `CheckIfThereAreChanges()` your script looks at the HEAD of each branch rather than the $(Build.SourceVersion) commit via the Build Pipeline variables here: https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#build-variables-devops-services

Is it possible to make an option / checkbox to populate the delta via this environment variable? If you give me some direction I also may be able to help this weekend

**Comment by shayki5** [comment]
Hi @ColinKrist, thanks for your input!
If I understand correctly, do you want to use `Build.SourceVersion` (that is commit ID) here: `?baseVersion=$($sourceBranch)`? instead of `$sourceBranch`.

**Comment by ColinKrist** [comment]
Precisely! The docs read that supplying a tag, branch name, or in this case the commit hash for `baseVersion` is an option: 
![image](https://user-images.githubusercontent.com/27971797/108659760-6dbabf80-7490-11eb-9aa6-c1b357b663a8.png)

Reading the docs and getting familiar with it again I have thought of a possible workaround that would require no script changes. I could technically provide the $(Build.SourceVersion) as the variable for the Source Branch in the YAML. I think it's possible since it looks like the script passes in sourceBranch mostly unmodified. 

I'll give it a go Monday morning, but a nice checkbox flow to auto pull that in the script would be awesome if that sounds desirable to you!

**Comment by shayki5** [comment]
@ColinKrist I don't think you can put `$(Build.SourceVersion)` in the source branch because when you create a PR  you need to supply a branch name and not a commit:

![image](https://user-images.githubusercontent.com/26063853/108675087-fc5e2980-74ee-11eb-9502-127071e3d330.png)

**Comment by ColinKrist** [comment]
Ah you're right - I wasn't looking closely enough at the create PR portion of your script. 

If we wanted to get the correct number of commits to attach - we'll have to add another param to the create PR endpoint and feed the array of commits to the endpoint:
![image](https://user-images.githubusercontent.com/27971797/108728796-81494300-74ef-11eb-8033-207bf9737d8d.png)

I think this is possible via this endpoint as it returns a `GitCommitRef[]` specifically:
https://docs.microsoft.com/en-us/rest/api/azure/devops/git/commits/get%20commits?view=azure-devops-rest-4.1#on-a-branch

With a little bit of local testing I was able to retrieve the right number of commits between master and the specific commit on develop by setting these URI params:
![image](https://user-images.githubusercontent.com/27971797/108729171-de44f900-74ef-11eb-913b-6c99d3e56d94.png)
![image](https://user-images.githubusercontent.com/27971797/108729203-e69d3400-74ef-11eb-8a4c-08d8ddf3fe24.png)

resulting url: https://dev.azure.com/medicalsolutions/Med%20Sol%20Tech/_apis/git/repositories/TextUsprocessor/commits?searchCriteria.itemVersion.version=master&searchCriteria.compareVersion.version=098fa8643e8257c3c3f2f630a67b602dde5d237c&searchCriteria.compareVersion.versionType=commit&api-version=4.1

**Comment by shayki5** [comment]
But did you success to create a PR only from those commits?

**Comment by ColinKrist** [comment]
Thank you for holding me to the thesis. I had to leave my thoughts there because I ran out of time. I just spent a bit more time on it and was unable to do so. Irritating that commits[] is part of the body, but does nothing.

I'm now exploring creating Ref/branch via a separate script YAML step and plugging that branch ref into your task instead:
![image](https://user-images.githubusercontent.com/27971797/109013969-52f76f00-7679-11eb-997e-cf75c3ac78dd.png)
https://docs.microsoft.com/en-us/rest/api/azure/devops/git/refs/update%20refs?view=azure-devops-rest-4.1#examples

I'm not sure if ref/branch creation is something you want in your task to handle a trunk based flow, so feel free to close this if you desire.
